Loader/SymfonyFixturesLoader.php

```diff
@@ -19,8 +19,31 @@
  */
 final class SymfonyFixturesLoader extends ContainerAwareLoader
 {
+    private $loadedFixtures = [];
+
+    /**
+     * @internal
+     */
+    public function addFixtures(array $fixtures)
+    {
+        // Store all loaded fixtures so that we can resolve the dependencies correctly.
+        foreach ($fixtures as $fixture) {
+            $this->loadedFixtures[get_class($fixture)] = $fixture;
+        }
+
+        // Now load all the fixtures
+        foreach ($this->loadedFixtures as $fixture) {
+            $this->addFixture($fixture);
+        }
+    }
+
     public function addFixture(FixtureInterface $fixture)
     {
+        $class = get_class($fixture);
+        if (!isset($this->loadedFixtures[$class])) {
```

Comment:
It might be worth throwing an exception if `$this->loadedFixtures[$class] !== $fixture` as that indicates that something fancy is going on.

Reply:
OK I will update tomorrow and also add tests for it.

Reply:
Fixed.

Reply:
This seems to let the test fail.... will investigate :)

Reply:
@alcaeus I think the tests are failing because https://github.com/doctrine/data-fixtures/pull/274 was merged in data-fixtures 1.3.0.

How can we handle this properly? I think we should require a higher data-fixtures version?

Reply:
Since `SymfonyFixturesLoader` now depends on `SymfonyFixturesLoader::createFixture`

Reply:
@weaverryan created a workaround to avoid bumping the dependency (which would implicitly bump the PHP requirement for the bundle as well, which they were trying to avoid). I'll take a look at it later today.